### ISSUE TITLE
Support for stateless container

### ISSUE THREAD

**Issue by bgwastu** [openblocks-dev/openblocks#155]
Does openblocks run on stateless container and works without database when the app deployed?

As always, thank you for building this great tool!

**Comment by neon-balcony** [comment]
Hey @bagaswastu , 

Thanks for the post, Openblocks is a platform for those built apps to run on, and Openblocks uses a MongoDB for fetching workspace/user/app/... data and a Redis for maintaining user sessions. So I would say Openblocks is **stateful** in nature. But it does not require your extra data storages if you're just using a single self-hosted instance with container built-in MongoDB and Redis.

**Comment by bgwastu** [comment]
Thank you for your response. 

I see that Openblocks uses supervisord to run multiple services in a single container. If I'm using my own Redis and Mongo services, do I still need to specify a volume in the Docker container? I'm planning to deploy Openblocks on Google Cloud Run, but since the container is stateless on Cloud Run, I'm worried that I won't be able to deploy without specifying a volume.

**Comment by neon-balcony** [comment]
I see, I think it should be OK without specifying volume in docker setup if you're using your own Redis and MongoDB. And you just need to specify environment variables for your MongoDB and Redis URI then.

**Comment by rhuanbarreto** [comment]
But MongoDB and Redis are still being started by suervisord if the environment variables are set, which can cause issues if you mount volumes in k8s (Specially in Azure Files that you need to set mount specific options to avoid having a crashed mongodb due to file permission issues for WiredTiger).

**Comment by neon-balcony** [comment]
@rhuanbarreto Got it, it seems that we can provide a _pure_ version Openblocks without built-in MongoDB and Redis. We've added it to our backlog and thanks for the info 😄

**Comment by dizzy-duality** [comment]
We would love this feature as wel!

**Comment by YoJayz** [comment]
Hi everyone, separate images of follwing services are available now:

- api-service: Backend service.
- node-service: Backend service.
- frontend: Frontend service.
- MongoDB: Used for persisting data of users, apps, data sources, etc.
- Redis: Used for maintaining user sessions, rate limiter, etc.

The detailed info about self-hosting with separate images is provided in this doc: https://docs.openblocks.dev/self-hosting#multi

Hope you find this helpful for stateless containers in cluster environment! 😃

**Comment by tboulogne** [comment]
hello @YoJayz,

I'm trying to use the multi-container and i face a strange issue in api-service :
![image](https://user-images.githubusercontent.com/6205275/226292951-929e806a-0ab5-443c-8149-0e07f734c27e.png)

Tried with environment variable REDIS_URI or REDIS_URL, same.

Any idea to solde please ?

Thanks for help.

Regards

**Comment by YoJayz** [comment]
> hello @YoJayz,
> 
> I'm trying to use the multi-container and i face a strange issue in api-service : ![image](https://user-images.githubusercontent.com/6205275/226292951-929e806a-0ab5-443c-8149-0e07f734c27e.png)
> 
> Tried with environment variable REDIS_URI or REDIS_URL, same.
> 
> Any idea to solde please ?
> 
> Thanks for help.
> 
> Regards

Hi @tboulogne 
For better understanding of this issue, would you please provide your masked REDIS_URI to us?

**Comment by tboulogne** [comment]
yes,

tried both :
```
REDIS_URL=redis://<user>:<password>@<service>:6379
REDIS_URI=redis://<user>:<password>@<service>:6379
```

**Comment by jerry-goodman** [comment]
REDIS_URL=redis://<user>:<password>@<service>:6379 
looks fine, can you connect to it successfully with other Redis clients?

**Comment by tboulogne** [comment]
@jerry-goodman yes i tried with redis-cli from app container and the connection to redis container is ok.

**Comment by tboulogne** [comment]
update:

@jerry-goodman @YoJayz 
Looking into the api-service container i found 2 files:
![image](https://user-images.githubusercontent.com/6205275/226457269-f9dcfb20-ee94-4fb9-9624-c804b0ea61c9.png)

The application-selfhost hold the parameters where the application not. (no mention of redis).

Normal behavior for this file ?

Thanks for help.

**Comment by YoJayz** [comment]
> @jerry-goodman yes i tried with redis-cli from app container and the connection to redis container is ok.

Sorry for the late reply @tboulogne 
Would you like to join our official [Discord](https://discord.com/invite/z5W2YHXdtt) server so that our engineer can have a **private chat** in detail with you?

**Comment by tboulogne** [comment]
hello @YoJayz just joined :-). Let me know.
